### ISSUE TITLE
ESlint failing to start due to React dependency?

### ISSUE THREAD

**Issue by memilanuk** [wesbos/eslint-config-wesbos#59]
I've done the walk-thru multiple times (including the part where I have to *un*install everything and start over again.

ESlint isn't doing *anything* to re-format malformed lines as shown in the video.  When I click on the 'ESlint' icon in the toolbar (which has to be enabled separately now, apparently) I get the lines shown below:

```
[Info  - 12:14:59 AM] ESLint server running in node v12.8.1
[Info  - 12:14:59 AM] ESLint server is running.
[Info  - 12:17:16 AM] ESLint library loaded from: C:\Users\memil\Downloads\Beginner Javascript\course files\node_modules\eslint\lib\api.js
Warning: React version was set to "detect" in eslint-plugin-react settings, but the "react" package is not installed. Assuming latest React version for linting.
[Info  - 12:17:18 AM] Error while loading rule 'react/jsx-no-bind': Cannot find module 'react' from 'C:\Users\memil\Downloads\Beginner Javascript\course files' Occurred while linting C:\Users\memil\Downloads\Beginner Javascript\course files\playground\some-FINISHED.js
[Info  - 12:21:19 AM] Error while loading rule 'react/display-name': Cannot find module 'react' from 'C:\Users\memil\Downloads\Beginner Javascript\course files' Occurred while linting C:\Users\memil\Downloads\Beginner Javascript\course files\playground\some-FINISHED.js
[Info  - 12:21:40 AM] Error while loading rule 'react/display-name': Cannot find module 'react' from 'C:\Users\memil\Downloads\Beginner Javascript\course files' Occurred while linting C:\Users\memil\Downloads\Beginner Javascript\course files\playground\some-FINISHED.js

```

After running `npm install react`,  I get a success message, but a number of other warnings about peerdeps (sorry, didn't capture those).

Anway, after installing react, which is an odd dependency since the Beginner Javascript course doesn't - as far as I can tell - use react or even touch on it, now I have ESlint about half-way working like shown in the video (#5).

**Comment by Asjas** [comment]
@memilanuk This eslint config tries to detect React because the config includes React, JavaScript and HTML formatting in one. It's not specific to only the Beginner JavaScript course 🙂

**Comment by memilanuk** [comment]
@Asjas I figured as much.  But I think the argument could be made that for a "No-Sweat" setup that is specifically called out in the Beginner course, to have it depend on React - and then not even mention that - might be a bit of an oversight.

**Comment by Asjas** [comment]
React doesn't have to be installed - Eslint will try to detect it and will move on if it can't detect it. 🙂

**Comment by Asjas** [comment]
Doing a bit of testing on Windows 10 it seems that the version `"eslint-plugin-react": "^7.14.2"` is causing the issue where it completely stops `eslint` if `react` isn't installed. Updating the `eslint-plugin-react` package to `7.20.6` fixes the issue and it runs `eslint` even if `react` isn't installed in a project.

I'll open a PR to update the `eslint-plugin-react` package to the latest version 🙂

**Comment by memilanuk** [comment]
So... is this supposed to be fixed now?

I went through and cleared everything out, and re-installed... and still getting react dependency errors:

```

[Info  - 4:21:50 PM] ESLint server running in node v12.8.1
[Info  - 4:21:50 PM] ESLint server is running.
[Info  - 4:21:52 PM] ESLint library loaded from: C:\Users\memil\Downloads\Beginner Javascript\course files\node_modules\eslint\lib\api.js
Warning: React version was set to "detect" in eslint-plugin-react settings, but the "react" package is not installed. Assuming latest React version for linting.
[Info  - 4:21:54 PM] Error while loading rule 'react/jsx-no-bind': Cannot find module 'react' from 'C:\Users\memil\Downloads\Beginner Javascript\course files' Occurred while linting C:\Users\memil\Downloads\Beginner Javascript\course files\playground\types.js
[Info  - 4:21:54 PM] Error while loading rule 'react/display-name': Cannot find module 'react' from 'C:\Users\memil\Downloads\Beginner Javascript\course files' Occurred while linting C:\Users\memil\Downloads\Beginner Javascript\course files\playground\types-FINISHED.js
[Info  - 4:22:03 PM] Error while loading rule 'react/display-name': Cannot find module 'react' from 'C:\Users\memil\Downloads\Beginner Javascript\course files' Occurred while linting C:\Users\memil\Downloads\Beginner Javascript\course files\playground\types.js
[Info  - 4:22:19 PM] Error while loading rule 'react/display-name': Cannot find module 'react' from 'C:\Users\memil\Downloads\Beginner Javascript\course files' Occurred while linting C:\Users\memil\Downloads\Beginner Javascript\course files\playground\types.js
[Info  - 4:22:21 PM] Error while loading rule 'react/display-name': Cannot find module 'react' from 'C:\Users\memil\Downloads\Beginner Javascript\course files' Occurred while linting C:\Users\memil\Downloads\Beginner Javascript\course files\playground\types.js
[Info  - 4:22:25 PM] Error while loading rule 'react/display-name': Cannot find module 'react' from 'C:\Users\memil\Downloads\Beginner Javascript\course files' Occurred while linting C:\Users\memil\Downloads\Beginner Javascript\course files\playground\types.js
[Info  - 4:22:27 PM] Error while loading rule 'react/display-name': Cannot find module 'react' from 'C:\Users\memil\Downloads\Beginner Javascript\course files' Occurred while linting C:\Users\memil\Downloads\Beginner Javascript\course files\playground\types.js
```

**Comment by Asjas** [comment]
@wesbos merged the pull request yesterday but I see he hasn't cut a new release yet, it's still on `0.0.22` on npm. 🙂

You can change `eslint-plugin-react` to `7.20.6` in the `package.json` file and then re-run the installation with `npm i` if you want to continue so long until Wes releases a new version of this config.

**Comment by Asjas** [comment]
@memilanuk Wes just released version `0.0.23` - I tested it on Windows 10 and it's now installing the updated packages. And I see that `eslint` now works as well 🙂
